third_party/move/move-compiler-v2/src/lib.rs

```diff
@@ -153,6 +155,79 @@ where
     Ok((env, annotated_units))
 }
 
+/// Run move compiler and print errors to given writer. Returns the set of compiled units.
+pub fn run_move_compiler_with_emitter(
+    mut emitter: Box<dyn Emitter>,
+    options: Options,
```

Comment:
Why the code duplication? Could we not create the emitter based on the message format in options?

Reply:
Got it, it's a draft, I'll remove it when someone confirms that the Emitter trait approach works.

Reply:
Done.